src/main/java/org/opensearch/knn/index/memory/NativeMemoryLoadStrategy.java

```diff
@@ -182,7 +157,7 @@ class TrainingLoadStrategy
             NativeMemoryLoadStrategy<NativeMemoryAllocation.TrainingDataAllocation, NativeMemoryEntryContext.TrainingDataEntryContext>,
             Closeable {
 
-        private static TrainingLoadStrategy INSTANCE;
+        private static volatile TrainingLoadStrategy INSTANCE;
```

Comment:
We need to make it volatile in Singleton pattern to avoid possible instruction order changes.

Reply:
lets add this as a java doc.

Reply:
will do in the next update